File: v6/sprefa-extract/tests/fixtures/go/docs.go
// docs.go: doc-facet parity fixture. Exercises v5's walk_go_docs: the
// contiguous line-comment block directly above a type spec (struct /
// interface / alias), function, or method declaration becomes a DocFact
// (go_leading_doc walks prev-sibling comments, requiring row adjacency; a
// blank line breaks the block). The documented entities re-exercise the
// ported facets (type/call/df) on doc-heavy input. ASCII-only so tree-sitter
// byte spans round-trip cleanly (parity is clean).

package docs

// Engine is a tiny string-bearing struct.
type Engine struct {
	name string
}

// Sizer reports a size.
type Sizer interface {
	Size() int
}

// Mode is an int alias.
type Mode int

// Trim returns its input unchanged.
func Trim(value string) string {
	return value
}

// MakeEngine builds an Engine from a name.
// A second doc line, to exercise the multi-line block walk.
func MakeEngine(name string) Engine {
	trimmed := Trim(name)
	engine := Engine{name: trimmed}
	return engine
}

// Mode picks the fast mode.
func (e *Engine) Mode() Mode {
	picked := Mode(0)
	return picked
}

File: v6/sprefa-extract/tests/fixtures/rust/docs.rs
// docs.rs: doc-facet parity fixture. Exercises v5's rust_docs_from: the
// doc-attribute (the desugared form of the triple-slash doc comment) on a
// struct/enum/trait/fn item or an impl method becomes a DocFact keyed by the
// entity's sym. Item kinds OUTSIDE that walk (const/static/type-alias) mint no
// doc row, and plain line comments are never docs (`apply` below). The
// documented entities re-exercise the ported facets (type/call/df/const) on
// doc-heavy input. ASCII-only so syn's char column equals the byte column
// (parity is clean).

/// A string-bearing engine.
pub struct Engine {
    name: String,
}

/// Operating modes.
pub enum Mode {
    Fast,
    Slow,
}

// A string const: the const facet fires, but const items get NO doc row.
pub const GREETING: &str = "hello";

/// Trims a value down.
pub fn trim(value: String) -> String {
    value
}

/// Builds an engine from a name.
pub fn make_engine(name: String) -> Engine {
    let trimmed = trim(name);
    let engine = Engine { name: trimmed };
    engine
}

impl Engine {
    /// Picks the fast mode.
    pub fn mode(&self) -> Mode {
        let picked = Mode::Fast;
        picked
    }
}

// Plain comments are not docs: no doc row for this fn.
pub fn apply(value: String) -> String {
    let func = |text: String| text;
    func(value)
}
